meshio/helpers.py

```diff
@@ -76,7 +76,24 @@
     '.vtk': 'vtk-binary',
     '.xdmf': 'xdmf',
     '.xmf': 'xdmf',
-    }
+}
+
+
+def check_filename(filename, file_format=None):
+    # Checks the input and output file name and format.
+    # https://stackoverflow.com/q/4843173/353337
+    assert isinstance(filename, str), 'No such file or directory: {}'.format(filename)
+
+    if not file_format:
+        # deduce file format from extension
+        extension = '.' + os.path.basename(filename).split(os.extsep, 1)[-1]
+        # check whether the file format is supported
+        if extension in _extension_to_filetype:
+            file_format = _extension_to_filetype[extension]
+        else:
+            assert False, ('Unknown file extension \'{}\' of \'{}\'.'
```

Comment:
Instead of `assert False`, do an
```python
assert extension in _extension_to_filetype, 'Error message bla'
```
before.

Reply:
Thanks for the comments. will incorporate them into the code.